test/framework/cluster_proxy.go

```diff
@@ -176,15 +176,17 @@ func (p *clusterProxy) GetClient() client.Client {
 	config := p.GetRESTConfig()
 
 	var c client.Client
+	var newClientErr error
 	err := wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
-		var newClientErr error
 		c, newClientErr = client.New(config, client.Options{Scheme: p.scheme})
 		if newClientErr != nil {
-			return false, newClientErr
+			return false, nil //nolint:nilerr
 		}
 		return true, nil
 	})
-	Expect(err).ToNot(HaveOccurred(), "Failed to get controller-runtime client")
+	errorString := "Failed to get controller-runtime client"
+	Expect(err).ToNot(HaveOccurred(), errorString)
+	Expect(newClientErr).ToNot(HaveOccurred(), errorString)
```

Comment:
I think this line is unreachable if the poll reaches the timeout because newClientError is != nil there will be an `ErrWaitTimeout` in `err`

Reply:
And reverse: If err is nil that is because `newClientErr` was nil

Reply:
Because `newClientErr` may be nil in the scenario when the initial attempt timed out before a response from `client.New` was returned we have to check `err`, but your comments point out that the order above should be reversed:

We should always check `newClientErr` first because its error object will have meaningful information about what went wrong establishing a client connection. But because `newClientErr` can be nil in a failure outcome, we need to check for `err` as well.

I'll switch the `Expect` order.

Reply:
Done, thx for your 👀 !